src/jit/lsra.cpp

```diff
@@ -1386,6 +1389,105 @@ bool LinearScan::isRegCandidate(LclVarDsc* varDsc)
     {
         return false;
     }
+
+    // Don't enregister if the ref count is zero.
```

Comment:
This section of code, and the above check for `!enregisterLocalVars` were extracted from `identifyCandidates()` so that `isRegCandidate` can be called from the assert in `Lowering`.